src/Sylius/Component/Promotion/Repository/PromotionCouponRepositoryInterface.php

```diff
@@ -43,4 +43,11 @@ public function countByCodeLength(int $codeLength): int;
      * @return PromotionCouponInterface|null
      */
     public function findOneByCodeAndPromotionCode(string $code, string $promotionCode): ?PromotionCouponInterface;
+
+    /**
+     * @param string $promotionCode
+     *
+     * @return iterable
+     */
+    public function findByPromotionCode(string $promotionCode): iterable;
```

Comment:
All find* repository methods returns array of all results, this one returns a paginator, we should mention it somewhere or change the method name at all (`createPaginatorForPromotion`?).

Reply:
Does it need to return a paginator? IIRC the paginator is applied automatically if a route is paginated?

Reply:
It works only when there's no repository method defined (here is) and for the criteria specified in routing. I'm not sure whether we can query by a nested field `promotion.code` using it.

Reply:
Ah, your right. In that case, I'd suggest putting paginator in the name 👍

Reply:
Btw. supporting relations in `criteria` looks like a useful feature.